api/C/include/csi_ll_pmp.h

```diff
@@ -34,33 +34,204 @@ typedef struct {
+ * with success, it will not be effective untill the core has been exit M-mode.
+ *
+ * @param idx: The index of PMP config entries
+ * @param flag_x: The excecutable flag of PMP config entries.
+ * @return : Status code
+ */
+csi_status_t csi_pmp_set_executable(unsigned int idx, bool flag_x);
+
+/*
+ * Get PMP enctry excecutable flag by index.
+ *
+ * This function only can be called in M-mode. If it is called
+ * in other mode, it may return an error or cause carsh. When this function is called
+ * with success, it will not be effective untill the core has been exit M-mode.
+ *
+ * @param idx: The index of PMP config entries
+ * @param flag_x: A pointer to the buffer which is used to output the excecutable flag
+ *                  of PMP config entries.
+ * @return : Status code
+ */
+csi_status_t csi_pmp_get_executable(unsigned int idx, bool *flag_x);
+
+/*
+ * Set PMP enctry readable flag by index.
+ *
+ * This function only can be called in M-mode. If it is called
+ * in other mode, it may return an error or cause carsh. When this function is called
+ * with success, it will not be effective untill the core has been exit M-mode.
+ *
+ * @param idx: The index of PMP config entries
+ * @param flag_r: The readable flag of PMP config entries.
+ * @return : Status code
+ */
+csi_status_t csi_pmp_set_readable(unsigned int idx, bool flag_r);
+
+/*
+ * Get PMP enctry readable flag by index.
+ *
+ * This function only can be called in M-mode. If it is called
+ * in other mode, it may return an error or cause carsh. When this function is called
+ * with success, it will not be effective untill the core has been exit M-mode.
+ *
+ * @param idx: The index of PMP config entries
+ * @param flag_r: A pointer to the buffer which is used to output the readable flag
+ *                  of PMP config entries.
+ * @return : Status code
+ */
+csi_status_t csi_pmp_get_readable(unsigned int idx, bool *flag_r);
+
+/*
+ * Set PMP enctry writeable flag by index.
  *
  * This function only can be called in M-mode. If it is called
  * in other mode, it may return an error or cause carsh. When this function is called
  * with success, it will not be effective untill the core has been exit M-mode.
  *
  * @param idx: The index of PMP config entries
- * @param base_addr: The base address of this range need to be configured
- * @param size: The size of this range need to be configured
- * @param attr: A point to the attribution of this range to be configured
- * @param enable: Eanble flag to indicate whether this ranged can be accessed or not
+ * @param flag_w: The writeable flag of PMP config entries.
  * @return : Status code
  */
-csi_status_t csi_mpu_config_region(unsigned int idx, unsigned long base_addr,
-                                   unsigned long size, mpu_region_attr_t *attr,
-                                   unsigned int enable);
+csi_status_t csi_pmp_set_writeable(unsigned int idx, bool flag_w);
```

Comment:
writable

Reply:
Hi Vincent，

I've fixed them. Pls help to check them agian

Thanks